core/trino-parser/src/main/java/io/trino/sql/tree/AssignmentStatement.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class AssignmentStatement
+        extends ControlStatement
+{
+    private final Identifier target;
+    private final Expression value;
+
+    public AssignmentStatement(Identifier target, Expression value)
```

Comment:
This constructor is not needed. There will always be a NodeLocation.

We use this form for AST elements that can be used in the IR and thus can be synthesized without a location by the optimizer.

Similar comment for all the other AST classes below.